### PR TITLE
Write a proper readme

### PR DESCRIPTION
**Pull request by i509VCB**
This is long overdue, so I have taken the time to write the start of a larger readme PR. This just covers the root readme.md to start

### CODE REVIEW

**Review by hYdos — APPROVED** [review]
yeef

File: README.md [hunk 1]
```diff
@@ -1,7 +1,94 @@
...
+## Contributing
+
+See something Fabric API doesn't support, a bug or something that may be useful? We welcome contributions to improve Fabric API.
+
+Check out [the Contributing guidelines](../CONTRIBUTING.md)*.
+
+\* The contributing guidelines are work in progress
+<!-- relative link will be fixed once as part of fabric repo, needs testing in fork -->
+
+<!--
```
Comment (CoolMineman) [inline README.md]: Why is this commented out?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,94 @@
...
 
-To setup a Fabric development environment, check out the [example fabric mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there.
+- Exposing functionality that is useful but difficult to access for many mods such as particles, biomes and dimensions
+- Adding events, hooks and APIs to improve interopability between mods.
+- Essential features such as registry synchronization and adding information to crash reports.
+- An advanced rendering API designed for compatibility with optimization mods and graphics overhaul mods.
+
+Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version independent mod loader that powers Fabric. Fabric API is a mod like any other fabric mod which requires Fabric Loader to be installed.
+
+For support and discussion for both developers and users, visit [the Fabric Discord server](https://discord.gg/v6v4pMv).
```
Comment (CoolMineman) [inline README.md:14]: Maybe Link [https://fabricmc.net/discuss/](https://fabricmc.net/discuss/)?

**Review by CoolMineman — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (i509VCB) [reply README.md:14]: The discuss page is actually a bit out of date since most development questions I see are on discord.

File: README.md (hunk 1 above)
Comment (i509VCB) [reply README.md]: I need to drop those comments

**Review by CoolMineman — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Way better than what we have, it can always be improved later if needed

**Review by Technici4n — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,7 +1,87 @@
...
+
+Check out [the Contributing guidelines](../CONTRIBUTING.md)*.
+
+\* The contributing guidelines are work in progress
+
+## Modules
+
+Fabric API is designed to be modular for ease of updating. This also has the advantage of splitting up the codebase into smaller chunks.
+
+Each module contains it's own `README.md`* explaining the module's purpose and additional info on using the module.
```
Comment (haykam821) [inline README.md]:
```suggestion
Each module contains its own `README.md`* explaining the module's purpose and additional info on using the module.
```

File: README.md [hunk 4]
```diff
@@ -1,7 +1,87 @@
...
+
+\* The readme for each module is being worked on, not every module has a readme at the moment
```
Comment (haykam821) [inline README.md]:
```suggestion
\* The README for each module is being worked on; not every module has a README at the moment
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,87 @@
...
+```groovy
+modImplementation "net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION"
+```
+
+**Kotlin DSL**
+```kotlin
+modImplementation("net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION")
+```
+
+Individual modules from Fabric API may be resolved via the `fabricApi` extension in loom for a smaller dependency footprint:
```
Comment (haykam821) [inline README.md]:
```suggestion
Individual modules from Fabric API may be resolved via the `fabricApi` extension in Loom for a smaller dependency footprint:
```

File: README.md [hunk 6]
```diff
@@ -1,7 +1,87 @@
...
+
+The downloaded jar file should be placed in your `mods` folder.
+
+## Getting Started With Development
+
+To setup a Fabric development environment, check out the [fabric example mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there. The Example mod should already include Fabric API.
+
+To include all of Fabric API in a mod development environment, add the following to your `dependencies` block in the gradle buildscript:
+
+**Groovy DSL**
```
Comment (haykam821) [inline README.md]:
```suggestion
### Groovy DSL

```

File: README.md [hunk 7]
```diff
@@ -1,7 +1,87 @@
...
+```
+
+**Kotlin DSL**
+```kotlin
+modImplementation("net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION")
+```
+
+Individual modules from Fabric API may be resolved via the `fabricApi` extension in loom for a smaller dependency footprint:
+
+**Groovy DSL**
```
Comment (haykam821) [inline README.md]:
```suggestion
### Groovy DSL

```

File: README.md [hunk 8]
```diff
@@ -1,7 +1,87 @@
...
+    "fabric-networking-v0"
+]
+
+// Add each module as a dependency
+apiModules.forEach {
+    modImplementation(fabricApi.module(it, FABRIC_API_VERSION))
+}
+```
+
+**Kotlin DSL**
```
Comment (haykam821) [inline README.md]:
```suggestion
### Kotlin DSL

```

File: README.md [hunk 9]
```diff
@@ -1,7 +1,87 @@
...
+To setup a Fabric development environment, check out the [fabric example mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there. The Example mod should already include Fabric API.
+
+To include all of Fabric API in a mod development environment, add the following to your `dependencies` block in the gradle buildscript:
+
+**Groovy DSL**
+```groovy
+modImplementation "net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION"
+```
+
+**Kotlin DSL**
```
Comment (haykam821) [inline README.md]:
```suggestion
### Kotlin DSL

```

File: README.md [hunk 10]
```diff
@@ -1,7 +1,87 @@
...
+
+## Using Fabric API
+
+To use fabric api, download Fabric API from [Curseforge](https://www.curseforge.com/minecraft/mc-mods/fabric-api) or [GitHub Releases](https://github.com/FabricMC/fabric/releases).
+
+The downloaded jar file should be placed in your `mods` folder.
+
+## Getting Started With Development
+
+To setup a Fabric development environment, check out the [fabric example mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there. The Example mod should already include Fabric API.
```
Comment (haykam821) [inline README.md]:
```suggestion
To setup a Fabric development environment, check out the [Fabric example mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there. The example mod already depends on Fabric API.
```

File: README.md [hunk 11]
```diff
@@ -1,7 +1,87 @@
...
+- Essential features such as registry synchronization and adding information to crash reports.
+- An advanced rendering API designed for compatibility with optimization mods and graphics overhaul mods.
+
+Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version independent mod loader that powers Fabric. Fabric API is a mod like any other fabric mod which requires Fabric Loader to be installed.
+
+For support and discussion for both developers and users, visit [the Fabric Discord server](https://discord.gg/v6v4pMv).
+
+## Using Fabric API
+
+To use fabric api, download Fabric API from [Curseforge](https://www.curseforge.com/minecraft/mc-mods/fabric-api) or [GitHub Releases](https://github.com/FabricMC/fabric/releases).
```
Comment (haykam821) [inline README.md]:
```suggestion
To use Fabric API, download it from [CurseForge](https://www.curseforge.com/minecraft/mc-mods/fabric-api) or [GitHub Releases](https://github.com/FabricMC/fabric/releases).
```

File: README.md [hunk 12]
```diff
@@ -1,7 +1,87 @@
...
-## Getting Started
+Fabric API is the library for essential hooks and interoperability mechanisms for Fabric mods. Examples include:
 
-To setup a Fabric development environment, check out the [example fabric mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there.
+- Exposing functionality that is useful but difficult to access for many mods such as particles, biomes and dimensions
+- Adding events, hooks and APIs to improve interopability between mods.
+- Essential features such as registry synchronization and adding information to crash reports.
+- An advanced rendering API designed for compatibility with optimization mods and graphics overhaul mods.
+
+Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version independent mod loader that powers Fabric. Fabric API is a mod like any other fabric mod which requires Fabric Loader to be installed.
```
Comment (haykam821) [inline README.md]:
```suggestion
Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version-independent mod loader that powers Fabric. Fabric API is a mod like any other Fabric mod which requires Fabric Loader to be installed.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Review by LambdAurora — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Player3324) [reply README.md:14]: We can take care of this link after the website is reworked.

File: README.md [hunk 13]
```diff
@@ -1,7 +1,91 @@
...
+modImplementation "net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION"
+```
+
+### Kotlin DSL
+
+```kotlin
+modImplementation("net.fabricmc.fabric-api:fabric-api:FABRIC_API_VERSION")
+```
+
+Individual modules from Fabric API may be resolved via the `fabricApi` extension in Loom for a smaller dependency footprint:
```
Comment (Player3324) [inline README.md]: This isn't very clear IMO, one may not recognize that "all of Fabric API" above means all modules and that the individual modules added here are related.

I'd change the wording to something like "To include the full Fabric API with all modules in a mod [...]" and "Alternatively, modules from Fabric API can be specified individually via the [...]"

File: README.md [hunk 14]
```diff
@@ -1,7 +1,91 @@
...
+- Essential features such as registry synchronization and adding information to crash reports.
+- An advanced rendering API designed for compatibility with optimization mods and graphics overhaul mods.
+
+Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version-independent mod loader that powers Fabric. Fabric API is a mod like any other Fabric mod which requires Fabric Loader to be installed.
+
+For support and discussion for both developers and users, visit [the Fabric Discord server](https://discord.gg/v6v4pMv).
+
+## Using Fabric API
+
+To use Fabric API, download it from [CurseForge](https://www.curseforge.com/minecraft/mc-mods/fabric-api) or [GitHub Releases](https://github.com/FabricMC/fabric/releases).
```
Comment (Player3324) [inline README.md:20]: I think this needs to describe the kind of use, i.e. playing with fabric mods vs using fabric api for developing mods.

File: README.md [hunk 15]
```diff
@@ -1,7 +1,91 @@
...
+    "fabric-lifecycle-events-v1",
+    "fabric-networking-v0"
+).forEach {
+    // Add each module as a dependency
+    modImplementation(fabricApi.module(it, FABRIC_API_VERSION))
+}
+```
+
+<!--Linked to gradle documentation on properties-->
+Gradle properties may be used to catalog dependency versions in a single `gradle.properties` file at the root of a project. More information is available [here](https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#declare_properties_in_gradle_properties_file).
```
Comment (Player3324) [inline README.md]: Difficult to follow IMO, could state something like "Instead of hardcoding version constants all over the build script, [...]"

**Review by Player3324 — APPROVED** [review]
Looks good overall, I have a few potential enhancements that can also be done at a later time though.

File: README.md [hunk 16]
```diff
@@ -13,17 +13,19 @@ Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (
...
 
 The downloaded jar file should be placed in your `mods` folder.
 
-## Getting Started With Development
+## Using Fabric API to develop mods
 
 To setup a Fabric development environment, check out the [Fabric example mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there. The example mod already depends on Fabric API.
 
-To include all of Fabric API in a mod development environment, add the following to your `dependencies` block in the gradle buildscript:
+To include the full Fabric API with all modules in a mod, add the following to your `dependencies` block in the gradle buildscript:
```
Comment (Player3324) [inline README.md]: The removal of development environment makes this tentatively wrong now, either "include [...] in the dev env" or "to use the full fabric api..."

**Review by Player3324 — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**